flocker/provision/_common.py

```diff
@@ -69,3 +72,74 @@ class Cluster(PRecord):
     @property
     def certificates_path(self):
         return self.certificates.directory
+
+
+class INode(Interface):
+    """
+    Interface for node for running acceptance tests.
+    """
+    address = InterfaceAttribute('Public IP address for node')
+    private_address = InterfaceAttribute('Private IP address for node')
+    distribution = InterfaceAttribute('distribution on node')
+
+    def get_default_username():
+        """
+        Return the username available by default on a system.
+
+        Some cloud systems (e.g. AWS) provide a specific username, which
+        depends on the OS distribution started.  This method returns
+        the username based on the node distribution.
+        """
+
+    def provision(package_source, variants):
+        """
+        Provision flocker on this node.
+
+        :param PackageSource package_source: The source from which to install
+            flocker.
+        :param set variants: The set of variant configurations to use when
+            provisioning
+        """
+
+    def destroy():
+        """
+        Destroy the node.
+        """
+
+    def reboot():
+        """
+        Reboot the node.
+
+        :return Effect:
+        """
+
+
+class IProvisioner(Interface):
+    """
+    A provisioner for creating nodes to run acceptance tests agasint.
+    """
+    default_size = InterfaceAttribute(
```

Comment:
It doesn't seem that this attribute is required to be part of the interface. It is only used by the implementation of `create_node`. Leave it out until it's actually needed by external code.